### ISSUE TITLE
Why does not Crypt check hmac?

### ISSUE THREAD

**Issue by yejune** [phalcon/cphalcon#13379]
Why does not Crypt check hmac?

When decrypting an encrypted character, entering a different key will return an incorrect decryption character.

```php
 $text = 'This is a secret text';

 $key       = 'le password';
 $encrypted = $crypt->encrypt($text, $key);

 $key       = 'le password2';
 echo $crypt->decrypt($encrypted, $key);
```

result:
```
���0�Pz�@��/x��'�t��
```

If hmac is checked and is not valid, an empty string must be returned.

**Comment by yejune** [comment]
Sorry. I had a conceptual story.

If it is not valid, it throws an exception.

**Comment by sergeyklay** [comment]
Fixed in the [`3.4.0` version](https://github.com/phalcon/cphalcon/releases/tag/v3.4.0). Feel free to open a new issue if the problem appears again. Thank you for contributing.
